edge/pkg/edged/edged_pods.go

```diff
@@ -725,9 +878,83 @@ func (e *edged) makeEnvironmentVariables(pod *v1.Pod, container *v1.Container, p
 	for k, v := range tmpEnv {
 		result = append(result, kubecontainer.EnvVar{Name: k, Value: v})
 	}
+
+	// Append remaining service env vars.
+	for k, v := range serviceEnv {
+		// Accesses apiserver+Pods.
+		// So, the master may set service env vars, or kubelet may.  In case both are doing
+		// it, we skip the key from the kubelet-generated ones so we don't have duplicate
+		// env vars.
+		// TODO: remove this next line once all platforms use apiserver+Pods.
+		if _, present := tmpEnv[k]; !present {
```

Comment:
Does it really need to judge it again since the envVar has already be removed at line 872?

Reply:
They are values in different fields.

Reply:
Don't think so.
I think `if _, present := tmpEnv[k]; !present` and `delete(serviceEnv, envVar.Name)` are same logic.
That's to merge the map `serviceEnv` into `tmpEnv`. Since you have already deleted the `tmpEnv` key from `serviceEnv`, the no key in `serviceEnv` presents in `tmpEnv`.

But `delete` may miss some key from `container.EnvFrom`.

So for the simplicity and clarity, I think we can safely delete the `delete(serviceEnv, envVar.Name)` and the duplicate comments.

Another proof: the same comment `TODO: remove this next line once all platforms use apiserver+Pod`

Reply:
Got it.